src/mon/OSDMonitor.cc

```diff
@@ -5072,6 +5072,20 @@ void OSDMonitor::get_pools_health(
           detail->push_back(s);
       }
     }
+
+    if (sum.num_objects > 0 && pool.application_metadata.empty()) {
```

Comment:
Ok, since these labels go in teh OSDMap, and since we only encode then when we set the require_osd_release to luminous, we can do the check on the first map that adds the flag in OSDMonitor::encode_pending().  Then it happens exactly once at the end of the upgrade.